FrameworkSystem/Client/BundleDeliveryClient.py

```diff
@@ -2,30 +2,29 @@
 """
 
 import os
+import io
 import tarfile
 import cStringIO
 
 from DIRAC import S_OK, gLogger
-from DIRAC.Core.DISET.RPCClient import RPCClient
+from DIRAC.Core.Base.Client import Client, createClient
 from DIRAC.Core.DISET.TransferClient import TransferClient
 from DIRAC.Core.Security import Locations, Utilities
+from DIRAC.Core.Utilities.File import mkDir
 from DIRAC.ConfigurationSystem.Client.Helpers.CSGlobals import skipCACheck
+
+
 __RCSID__ = "$Id$"
 
 
-class BundleDeliveryClient(object):
+@createClient('Framework/BundleDelivery')
+class BundleDeliveryClient(Client):
 
-  def __init__(self, rpcClient=False, transferClient=False):
-    self.rpcClient = rpcClient
+  def __init__(self, transferClient=False, **kwargs):
+    Client.__init__(self, **kwargs)
```

Comment:
you also need to call `self.setServer`. The `url` parameter passed to the constructor of `Client` is useless :-(

Reply:
Done thanks.